03_myfavorite/src/Components/GamePage.jsx

```diff
@@ -0,0 +1,118 @@
+import { React, useEffect, useState } from 'react';
+import Card from './Card';
+import ResultPage from './ResultPage';
+import 네오 from '../img/네오.jpg';
+import 라이언 from '../img/라이언.jpg';
+import 무지 from '../img/무지.jpg';
+import 어피치 from '../img/어피치.jpg';
+import 제이지 from '../img/제이지.jpg';
+import 춘식이 from '../img/춘식이.jpg';
+import 튜브 from '../img/튜브.jpg';
+import 프로도 from '../img/프로도.jpg';
+import vs from '../img/vs.png';
+import styled from 'styled-components';
+
+const gameInfo = [
+  { name: '네오', src: 네오 },
+  { name: '라이언', src: 라이언 },
+  { name: '무지', src: 무지 },
+  { name: '어피치', src: 어피치 },
+  { name: '제이지', src: 제이지 },
+  { name: '춘식이', src: 춘식이 },
+  { name: '튜브', src: 튜브 },
+  { name: '프로도', src: 프로도 },
+];
+
+function GamePage() {
+  const [leftCharacters, setLeftCharacters] = useState(gameInfo);
+  const [winnersInfo, setWinnersInfo] = useState([]);
+  const [currentRound, setCurrentRound] = useState(1);
+  const [rounds, setRounds] = useState(4);
+  const [winner, setWinner] = useState({});
+
+  function handleCardClick(friend) {
+    if (rounds === 1) {
+      setWinner(friend);
+    }
+    setLeftCharacters(leftCharacters.slice(2));
+    setWinnersInfo((prev) => [...prev, friend]);
+    setCurrentRound((prev) => prev + 1);
+  }
+
+  useEffect(() => {
+    if (leftCharacters.length === 0) {
+      setLeftCharacters(winnersInfo);
+      setWinnersInfo([]);
+      setCurrentRound(1);
+      setRounds((prev) => prev / 2);
+    }
+  }, [leftCharacters]);
+
+  if (rounds === 0.5) {
+    return <ResultPage winner={winner} />;
+  }
```

Comment:
```suggestion
  useEffect(() => {
    if (rounds === 0.5) {
      ...
    }
  }, [rounds]);
```

지금 비록 state가 바뀌어서 컴포넌트가 리렌더링 되고, 해당하는 조건문을 확인할테지만,
state가 바뀌는 것을 observe 하려면 `useEffect` 로 확인해주는 게 좋아

setState 구문이 비동기로 처리되는 구문이라 로직이 꼬일 수 있을 것 같다는 생각이 드네!